### ISSUE TITLE
Regression test failure with PG 15.1: tests/test_connection.py: AssertionError: [] is not true

### ISSUE THREAD

**Issue by df7cb** [psycopg/psycopg2#1521]
It looks that #1506 is back - Pycopg2 2.9.5, PostgreSQL 15.1, Debian unstable/amd64:

```
FAIL: test_ssl_attribute (tests.test_connection.TestConnectionInfo)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/cbe/projects/postgresql/psycopg2/psycopg2/.pybuild/cpython3_3.10_psycopg2/build/tests/test_connection.py", line 1925, in test_ssl_attribute
    self.assert_(attribs)
AssertionError: [] is not true
```

The culprit seems to be postgres/postgres@cae4688ce8.

**Comment by dvarrazzo** [comment]
This area is being worked up quite a lot. Could you please suggest how to fix these testt in a way that won't be broken by upstream in two weeks? Should we drop them?

**Comment by df7cb** [comment]
TBH I don't really know. Given it's a 15.1 change, it's probably less likely that it will happen again in a higher minor release. I'd probably update to the current format, plus whatever the old releases produced, and forget that 15.0 existed.

**Comment by dvarrazzo** [comment]
Actually I don't see the relation between that changeset and this error.

The test failing

https://github.com/psycopg/psycopg2/blob/e8d92b74fd4b219cb4725cb906c526f3f2dce31d/tests/test_connection.py#L1923-L1925

relates to the function `PQsslAttributeNames()`, which doesn't seem touched by the changeset you mention. I don't know how to reproduce getting an empty list from that function:
```python
>>> import psycopg2

>>> psycopg2._psycopg.libpq_version()
150001

>>> conn = psycopg2.connect("")
>>> conn.info.ssl_in_use
False

>>> conn.info.ssl_attribute_names
['library', 'key_bits', 'cipher', 'compression', 'protocol']
```

I get a list even with a broken connection.

How is the psycopg2 in your tests built? Maybe with a libpq with no ssl support?

**Comment by df7cb** [comment]
Hmm, the tests are passing with 15.x now, so I'll just close this for now.

For PG16, see #1582.

**Comment by dvarrazzo** [comment]
Assumed fixed by #1582. If this is a problem for 15.1 only it's not worth being fixed.
